radix-engine/src/system/system.rs

```diff
@@ -1215,6 +1143,42 @@ where
 
         Ok(enabled)
     }
+
+    fn get_module_object_info(
+        &mut self,
+        node_id: &NodeId,
+        object_module_id: ObjectModuleId,
+    ) -> Result<ObjectInfo, RuntimeError> {
+        let object_info = self.get_object_info(node_id)?;
+        let module_object_info = match object_module_id {
+            ObjectModuleId::Main => object_info,
+            ObjectModuleId::Metadata | ObjectModuleId::Royalty | ObjectModuleId::AccessRules => {
+                let version =
+                    if let Some(version) = object_info.module_versions.get(&object_module_id) {
+                        version.clone()
+                    } else {
+                        return Err(RuntimeError::SystemError(
+                            SystemError::ObjectModuleDoesNotExist(object_module_id),
+                        ));
+                    };
+
+                // Modules outside of Main do not have their own modules
+                ObjectInfo {
+                    global: true,
+                    main_blueprint_id: object_module_id.static_blueprint().unwrap(),
```

Comment:
Following the above comment, the `main_blueprint_id` renaming doesn't look necessary.

In fact, it's actually confusing, because when the callee is an `Owned<Metadata>`, the `main_blueprint_id` is `metadata`.

```rust
MethodActor {
    node_id: NodeId,
    module_id: ObjectModuleId,
    ident: String,
    object_info: ObjectInfo {
         global: bool,
         blueprint_id: BlueprintId,
         modules: BTreeMap<ObjectModuleId, BlueprintVersion>  // module blueprint id statically inferred
    }
}
```